File: project/app/src/main/java/org/owntracks/android/injection/modules/android/FragmentModules/IntroFragmentModule.java
package org.owntracks.android.injection.modules.android.FragmentModules;


import androidx.fragment.app.Fragment;

import org.owntracks.android.injection.scopes.PerFragment;
import org.owntracks.android.ui.welcome.intro.IntroFragment;

import dagger.Binds;
import dagger.Module;

@Module(includes = BaseSupportFragmentModule.class)
public abstract class IntroFragmentModule {

    @Binds
    @PerFragment
    abstract Fragment bindSupportFragment(IntroFragment f);
}

File: project/app/src/main/java/org/owntracks/android/ui/welcome/intro/IntroFragmentModule.kt
package org.owntracks.android.ui.welcome.intro

import androidx.fragment.app.Fragment
import dagger.Binds
import dagger.Module
import org.owntracks.android.injection.scopes.PerFragment

@Module
abstract class IntroFragmentModule {
    @Binds
    @PerFragment
    abstract fun bindSupportFragment(f: IntroFragment?): Fragment?
}